src/Assignments/ProblemDetails.tsx

```diff
@@ -75,14 +82,16 @@ export const ProblemDetails: React.FC<ProblemDetailsProps> = ({
                         >
                             {(currentMoment: moment.Moment) => {
                                 let message = '';
-                                if (currentMoment.isBefore(endDate)) {
+                                if (currentMoment.isBefore(endDate) && (_.isNil(attemptsRemaining) || attemptsRemaining > 0)) {
                                     message = `Due ${endDate.fromNow()}`;
                                 } else if (currentMoment.isBefore(deadDate)) {
                                     message = `Partial credit expires ${deadDate.fromNow()}`;
                                 } else if (currentMoment.isBefore(solutionsMoment)) {
                                     message = `Solutions available ${solutionsMoment.fromNow()}`;
+                                } else if (isAssessment && (attemptsRemaining === 0 || isClosed)) {
+                                    message = 'is completed';
                                 } else {
-                                    message = 'Past due';
+                                    message = (isAssessment) ? 'Time expired' : 'Past due';
```

Comment:
Does it make sense to just always make it `Time expired`
I don't think it really matters either way

Reply:
Well, the exam itself isn't necessarily "past due" -- perhaps this could be more elaborate? 
Perhaps "Time expired _for this version_"?

Reply:
Hmm I think it's fine for now but I do like that